source/solvers/mf_navier_stokes_operators.cc

```diff
@@ -877,3 +903,394 @@ NavierStokesSUPGPSPGOperator<dim, number>::local_evaluate_residual(
 
 template class NavierStokesSUPGPSPGOperator<2, double>;
 template class NavierStokesSUPGPSPGOperator<3, double>;
+
+/**
+ * @brief This function performs a cell integral, i.e., the jacobian of the discretization
+ * of the transient Navier-Stokes equations with SUPG PSPG stabilization in a
+ * cell batch. The equations are given as follows: (q,∇δu) + (v,∂t δu) +
+ * (v,(u·∇)δu) + (v,(δu·∇)u) - (∇·v,δp) + ν(∇v,∇δu) (Weak form jacobian)
+ * + (∂t δu +(u·∇)δu + (δu·∇)u + ∇δp - ν∆δu)τ·∇q (PSPG Jacobian)
+ * + (∂t δu +(u·∇)δu + (δu·∇)u + ∇δp - ν∆δu)τu·∇v (SUPG Part 1)
+ * + (∂t u +(u·∇)u + ∇p - ν∆u - f )τδu·∇v (SUPG Part 2)
```

Comment:
I guess you could compute the content of the bracket and after that multiply it with `u`/`δu`.

Reply:
Yes, we could... we did this because there are so many terms and we wanted to keep it readable. Do you think it would be better to change it to improve performance?